test/common_utils.py

```diff
@@ -492,6 +492,28 @@ def get_num_channels(color_space):
     return num_channels
 
 
+def make_image(
+    spatial_size, *, color_space="RGB", batch_dims=(), dtype=None, device="cpu", memory_format=torch.contiguous_format
```

Comment:
Do we really need to call this `spatial_size`? Is `size` not good enough? We use `size` absolutely everywhere.

(We've had discussions in the past about whether `spatial_size` was even relevant for bboxes, and decided to not address due to lack of time.)

Reply:
> Do we really need to call this `spatial_size`? Is `size` not good enough? We use `size` absolutely everywhere.

That is the term we (before your time) decided to use. For example, the main helper function to extract the value from all our supported objects is

https://github.com/pytorch/vision/blob/2d4484fba0f45637f68adadf5a056a6147642aa4/torchvision/transforms/v2/functional/_meta.py#L117

Our reason for this was twofold:

1. Calling it `image_size` excludes the use case of videos
2. Just calling it `size` can lead to ambiguities for bounding boxes, since for them the value is independent of the actual tensor shape. For all other datapoints `spatial_size` and `.shape` are tightly coupled.

Thus, I've opted to call it `spatial_size` here as well. I'm ok with changing it to a better name if we find one, if we reflect that in the actual API as well.

> decided to not address due to lack of time

That is not what I remember. We left it as is because all alternatives were way more complex. Just as a quick example: how would you call a dispatcher with a bounding box if the spatial size (or whatever we might call it) is not stored on the object itself?

Reply:
You can see https://github.com/pytorch/vision/issues/7217 and the links there

Reply:
> Just calling it size can lead to ambiguities for bounding boxes

But how is this ambiguity resolved now that everything is called `spatial_size`? `spatial_size` refers to the size of an image and for a bbox it's the `spatial_size` of its image. This is literally the same situation as when `size` was used for everything?

I understand the need to distinguish between the size of the image of a bbox and the size of the bbox. But when everythign is called spatial_size, I don't see how we solved our issue

Reply:
The term "size" in PyTorch land is equivalent to the shape

```pycon
>>> t = torch.empty(2, 3, 4)
>>> t.shape
torch.Size([2, 3, 4, 5])
>>> t.size()
torch.Size([2, 3, 4, 5])
```

Meaning, calling everything `spatial_size` removes the ambiguity that we necessarily talk about the `.shape` of the tensor. It does not mean that `spatial_size` cannot be coupled with the shape, but it doesn't have to be.

Reply:
The term "size" in torchvision (e.g. in `Resize()` or the past `get_image_size()` etc.) refers to `H, W`. It's strongly tied to `.shape` and by extension to `.size()`, but it's not exactly that. There is no ambiguity about the use of "size" in torchvision.

I also don't think this argument stands considering we wouldn't have used the term `spatial_size` if it weren't for bboxes.

Reply:
Summary after further offline discussion in https://github.com/pytorch/vision/pull/7717#issuecomment-1621519282.